src/CommunityToolkit.Maui/Essentials/FileFolderDialog.android.cs

```diff
@@ -0,0 +1,435 @@
+﻿using System.ComponentModel;
+using Android.App;
+using Android.Content;
+using Android.Graphics;
+using Android.Util;
+using Android.Views;
+using Android.Widget;
+using Button = Android.Widget.Button;
+using Color = Android.Graphics.Color;
+using Environment = Android.OS.Environment;
+using File = Java.IO.File;
+using IOException = Java.IO.IOException;
+using View = Android.Views.View;
+
+namespace CommunityToolkit.Maui.Essentials;
+
+enum FileSelectionMode
+{
+	FileOpen,
+	FileSave,
+	FolderChoose,
+	FileOpenRoot,
+	FileSaveRoot,
+	FolderChooseRoot
+}
+
+sealed class FileFolderDialog : IDisposable
+{
+	readonly Context? context;
```

Comment:
`context` will never be null because it is assigned in the constructor

```suggestion
	readonly Context context;
```